crates/turborepo-globwalk/src/lib.rs

```diff
@@ -284,6 +282,23 @@ fn collapse_path(path: &str) -> Option<(Cow<str>, usize)> {
     }
 }
 
+fn glob_with_contextual_error<S: AsRef<str>>(raw: S) -> Result<Glob<'static>, WalkError> {
+    let raw = raw.as_ref();
+    Glob::new(raw)
+        .map(|g| g.into_owned())
+        .map_err(|e| WalkError::BadPattern(raw.to_string(), e))
+}
+
+pub(crate) fn any_with_contextual_error(
+    precompiled: Vec<Glob<'static>>,
+    text: Vec<String>,
+) -> Result<wax::Any<'static>, WalkError> {
+    wax::any(precompiled).map_err(|e| {
+        let text = text.iter().join(",");
+        WalkError::BadPattern(text, e)
+    })
+}
+
```

Comment:
So this is totally up to you, but if you wanted to avoid these helper functions here's a way to achieve something like anyhow's `with_context` method:
```
trait GlobContext<T>: Sized {
    fn result(self) -> Result<T, BuildError>;

    fn with_context<F>(self, f: F) -> Result<T, WalkError>
    where F: FnOnce() -> String
    {
        self.result().map_err(|e| WalkError::BadPattern(f(), e))
    }
}

impl<T> GlobContext<T> for Result<T, BuildError> {
    fn result(self) -> Result<T, BuildError> {
        self
    }
}
```

You can then replace the uses of `any_with_contextual_error(pattern, text)` with `wax::any(pattern).with_context(|| text.join(","))` (as long as `GlobContext` is within scope)

`glob_with_contextual_error` uses go back to `Glob::new(raw).with_context(|| raw.to_string())`

Reply:
I'm inclined to leave as-is and then do this for real when we're not in FFI-land. Ideally this would tie all the way back to the `turbo.json` line where the glob came from (or workspace glob or whatever). I understand people use [miette](https://docs.rs/miette/latest/miette/) for this kind of thing.